File: test/qtsocket/myserver.cpp
#include <QTcpServer>
#include <QTcpSocket>
#include <QHostAddress>
#include <QByteArray>
#include <QDebug>
#include <QObject>

class MyServer : public QTcpServer {
    Q_OBJECT

public:
    MyServer(QObject *parent = nullptr) : QTcpServer(parent) {
        qDebug() << "MyServer - initialized";
    }

protected:
    void incomingConnection(qintptr socketDescriptor) override {
        qDebug() << "New connection received with descriptor" << socketDescriptor;
        emit clientConnection(socketDescriptor);
    }

signals:
    void clientConnection(qintptr);
};

File: test/qtsocket/myserver.py
from PyQt5.QtNetwork import QTcpServer, QHostAddress
from PyQt5.QtCore import pyqtSignal

class MyServer(QTcpServer):
    clientConnection = pyqtSignal('qintptr')
    
    def __init__(self, parent=None):
        super().__init__(parent)
        print("MyServer - initialized")
    
    def incomingConnection(self, intptr):
        self.clientConnection.emit(intptr)